apps/frontend/src/components/chat/ChatPane.tsx

```
@@ -1,12 +1,44 @@
+    const { scrollTop, scrollHeight, clientHeight } = scrollContainerRef.current;
+    return scrollHeight - scrollTop - clientHeight < 100; // Within 100px of bottom
+  };
+
+  useEffect(() => {
+    // Only auto-scroll if user is near the bottom or this is the first message
+    if (currentThread?.messages.length === 1 || isNearBottom()) {
+      scrollToBottom();
+    }
+  }, [currentThread?.messages]);
+
+  useEffect(() => {
```

Biome (1.9.4)

[error] This hook does not specify all of its dependencies: isNearBottom

This dependency is not specified in the hook dependency list.


(lint/correctness/useExhaustiveDependencies)